### PR TITLE
[BIG tenor] Adjust tolerance for layer_norm

### PR DESCRIPTION
**Pull request by hxzd5568**
1. layer_norm 的实现是正确的。源码实现和torch 一致。
2. 误差较大原因是reduce过程会放大误差，通过缩小reduce 维度可以改正错误。
     如  paddle.nn.functional.layer_norm(Tensor([10361, 206, 1024],"float32"), 1024, weight=Tensor([1024],"float32"), bias=Tensor([1024],"float32"), epsilon=1e-05, ) -> accuracy error
     改为 paddle.nn.functional.layer_norm(Tensor([103610, 26, 1024],"float32"), 1024, weight=Tensor([1024],"float32"), bias=Tensor([1024],"float32"), epsilon=1e-05, ) -> pass
3. 增加精度可以缩小误差。而不同精度是同一套模板， 故没有逻辑问题。

pytorch:
  layer_norm_cuda
  layer_norm_backward_cuda
  pytorch/aten/src/ATen/native/cuda/layer_norm_kernel.cu
     ->vectorized_layer_norm_kernel_impl. (line 211)

paddle:
  Paddle/paddle/phi/kernels/funcs/layer_norm_impl.cu.h

### CODE REVIEW

**Review by cangtianhuang — APPROVED** [review]
LGTM
